src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -341,5 +365,110 @@ std::string GcsResourceManager::ToString() const {
   return ostr.str();
 }
 
+void GcsResourceManager::HandleGetClusterResourceState(
+    rpc::autoscaler::GetClusterResourceStateRequest request,
+    rpc::autoscaler::GetClusterResourceStateReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  // TODO: last_seen_cluster_resource_state_version not needed?
+  RAY_CHECK(request.last_seen_cluster_resource_state_version() <=
+            cluster_resource_state_version_);
+
+  reply->set_last_seen_autoscaler_state_version(last_seen_autoscaler_state_version_);
+  reply->set_cluster_resource_state_version(cluster_resource_state_version_);
+
+  GetNodeStates(reply);
+  GetPendingResourceRequests(reply);
+  GetPendingGangResourceRequests(reply);
+  GetClusterResourceConstraints(reply);
+
+  // QQ: should we use GCS_RPC_SEND_REPLY
```

Comment:
I think it is better not using that one. This will make the client side error handling complicated